### PR TITLE
[BEAM-2136] Use Thread context ClassLoader as parent in AvroCoderTest

### PR DESCRIPTION
**Pull request by kennknowles**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

This restores signal when maven is run with `-DforkCount=0`.

R: @lukecwik @aviemzur

### CODE REVIEW

**Comment by kennknowles** [comment]
run dataflow validatesrunner

**Comment by kennknowles** [comment]
Tests have passed, please take a look.

**Comment by kennknowles** [comment]
Actually, as per contribution policy, I am going to merge this to fix the postcommit signal. Please take a look at your earliest convenience.

File: sdks/java/core/src/test/java/org/apache/beam/sdk/coders/AvroCoderTest.java [hunk 1]
```diff
@@ -193,8 +198,10 @@ public void testAvroCoderEncoding() throws Exception {
    */
   @Test
   public void testTwoClassLoaders() throws Exception {
-    ClassLoader loader1 = new InterceptingUrlClassLoader();
-    ClassLoader loader2 = new InterceptingUrlClassLoader();
+    ClassLoader loader1 =
+        new InterceptingUrlClassLoader(Thread.currentThread().getContextClassLoader());
```
Comment (lukecwik) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/coders/AvroCoderTest.java]: Why isn't this ReflectHelpers.findClassLoader()?

**Pull request merged**